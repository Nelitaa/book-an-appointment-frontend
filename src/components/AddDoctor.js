import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux'; //
import { createDoctor } from '../redux/doctors';
import '../styles/add-doctor.css';

const AddDoctor = () => {
  const [name, setName] = useState('');
  const [specialization, setSpecialization] = useState('');
  const [city, setCity] = useState('');
  const [fee, setFee] = useState('');
  const [photo, setPhoto] = useState('');
  const [experience, setExperience] = useState('');
  const [message, setMessage] = useState('');

  const dispatch = useDispatch();

  const handleSubmit = async () => {
    try {
      const response = await dispatch(createDoctor({
        name,
        specialization,
        city,
        fee,
        photo,
        experience,
      }));
      console.log('Added doctor successfully:', response);
      console.log(response);
      // Update message state with the success message from response
      setMessage('Added doctor successfully' || '');
    } catch (error) {
      console.error('Add doctor failed:', error);
      setMessage(error.message || '');
    }
  };

  return (
    <div className="add-doctor-form">
      <p>{message}</p>
      <p className="add-doctor-title">Add a doctor</p>
      <div className="inputs">
        <input className="name-doctor-input" type="text" placeholder="Name" value={name} onChange={(e) => setName(e.target.value)} />
        <input className="specialization-input" type="text" placeholder="Specialization" value={specialization} onChange={(e) => setSpecialization(e.target.value)} />
        <input className="city-input" type="text" placeholder="City" value={city} onChange={(e) => setCity(e.target.value)} />
        <input className="fee-input" type="text" placeholder="Charging fee $" value={fee} onChange={(e) => setFee(e.target.value)} />
        <input className="photo-input" type="text" placeholder="Photo Link" value={photo} onChange={(e) => setPhoto(e.target.value)} />
        <input className="experience-input" type="text" placeholder="Years of experience" value={experience} onChange={(e) => setExperience(e.target.value)} />
        <button className="add-doctor-button" type="button" onClick={handleSubmit}>Add doctor</button>
        <Link to="/main" className="link back-doctors">Back to Doctors</Link>
      </div>
    </div>
  );
};

export default AddDoctor;
