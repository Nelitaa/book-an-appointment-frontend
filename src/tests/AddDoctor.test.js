import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import AddDoctor from '../components/AddDoctor';

it('Add doctor component should render correctly', () => {
  const result = render(
    <React.StrictMode>
      <Router>
        <Provider store={store}>
          <AddDoctor />
        </Provider>
      </Router>
    </React.StrictMode>,
  );
  expect(result).toMatchSnapshot();
});
