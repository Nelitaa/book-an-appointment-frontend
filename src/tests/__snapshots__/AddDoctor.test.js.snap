// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add doctor component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="add-doctor-form"
      >
        <p />
        <p
          class="add-doctor-title"
        >
          Add a doctor
        </p>
        <div
          class="inputs"
        >
          <input
            class="name-doctor-input"
            placeholder="Name"
            type="text"
            value=""
          />
          <input
            class="specialization-input"
            placeholder="Specialization"
            type="text"
            value=""
          />
          <input
            class="city-input"
            placeholder="City"
            type="text"
            value=""
          />
          <input
            class="fee-input"
            placeholder="Charging fee $"
            type="text"
            value=""
          />
          <input
            class="photo-input"
            placeholder="Photo Link"
            type="text"
            value=""
          />
          <input
            class="experience-input"
            placeholder="Years of experience"
            type="text"
            value=""
          />
          <button
            class="add-doctor-button"
            type="button"
          >
            Add doctor
          </button>
          <a
            class="link back-doctors"
            href="/main"
          >
            Back to Doctors
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="add-doctor-form"
    >
      <p />
      <p
        class="add-doctor-title"
      >
        Add a doctor
      </p>
      <div
        class="inputs"
      >
        <input
          class="name-doctor-input"
          placeholder="Name"
          type="text"
          value=""
        />
        <input
          class="specialization-input"
          placeholder="Specialization"
          type="text"
          value=""
        />
        <input
          class="city-input"
          placeholder="City"
          type="text"
          value=""
        />
        <input
          class="fee-input"
          placeholder="Charging fee $"
          type="text"
          value=""
        />
        <input
          class="photo-input"
          placeholder="Photo Link"
          type="text"
          value=""
        />
        <input
          class="experience-input"
          placeholder="Years of experience"
          type="text"
          value=""
        />
        <button
          class="add-doctor-button"
          type="button"
        >
          Add doctor
        </button>
        <a
          class="link back-doctors"
          href="/main"
        >
          Back to Doctors
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
